executor/builder.go

```diff
@@ -2230,7 +2230,7 @@ func (builder *dataReaderBuilder) buildProjectionForIndexJoin(ctx context.Contex
 	if int64(v.StatsCount()) < int64(builder.ctx.GetSessionVars().MaxChunkSize) {
 		e.numWorkers = 0
 	}
-	err = e.Open(ctx)
+	err = e.open(ctx)
```

Comment:
Can we try to refine the test case in the issue, and ensure IndexMergeJoin can be chosen?